### PR TITLE
Fix: local_window_size ignored when mask is None in dot_product_attention

### LINKED ISSUE #32300: jax.nn.dot_product_attention ignores local_window_size when mask is not present
### Description

Code that demonstrates this (run in a TPU Colab):

```python
import jax
import jax.numpy as jnp

BATCH_SIZE = 32
T_q = 2048
T_kv = 2080
NUM_HEADS = 8
HEAD_DIM = 64

key = jax.random.PRNGKey(42)
q_key, k_key, v_key, m_key = jax.random.split(key, 4)
q = jax.random.normal(
    q_key, (BATCH_SIZE, T_q, NUM_HEADS, HEAD_DIM), dtype=jnp.bfloat16
)
k = jax.random.normal(
    k_key, (BATCH_SIZE, T_kv, NUM_HEADS, HEAD_DIM), dtype=jnp.bfloat16
)
v = jax.random.normal(
    v_key, (BATCH_SIZE, T_kv, NUM_HEADS, HEAD_DIM), dtype=jnp.bfloat16
)

jitted_attention = jax.jit(
    lambda q, k, v, m, l: jax.nn.dot_product_attention(
        q, k, v, mask=m, local_window_size=(0, l - 1)
    ),
    static_argnums=4,
)

print(
    "Attention outputs match for different window sizes (32 vs 1) with no mask:",
    jnp.all(
        jitted_attention(q, k, v, None, 32)
        == jitted_attention(q, k, v, None, 1)
    ),
)

random_mask = jax.random.uniform(m_key, (BATCH_SIZE, 1, T_q, T_kv)) > 0.5

print(
    "Attention outputs match for different window sizes (32 vs 1) with a mask:",
    jnp.all(
        jitted_attention(q, k, v, random_mask, 32)
        == jitted_attention(q, k, v, random_mask, 1)
    ),
)
```

This code results in:
```
Attention outputs match for different window sizes (32 vs 1) with no mask: True
Attention outputs match for different window sizes (32 vs 1) with a mask: False
```

The culprit seems to be [this line](https://github.com/jax-ml/jax/blob/a589b31e20745d7ce7153d8fa8d7f815418d4dd4/jax/_src/nn/functions.py#L896) which prevents the local mask to get built [here](https://github.com/jax-ml/jax/blob/a589b31e20745d7ce7153d8fa8d7f815418d4dd4/jax/_src/nn/functions.py#L910).

### System info (python version, jaxlib version, accelerator, etc.)

```
jax:    0.8.0
jaxlib: 0.8.0
numpy:  2.3.3
python: 3.12.11 (stable, redacted, redacted) [Clang 9999.0.0 (fc44a4fcd3c54be927c15ddd9211aca1501633e7)]
device info: TPU v3-8, 8 local devices"
process_count: 1
platform: uname_result(system='Linux', node='6896a6a54d5a17c6-36316f7d32d.borgtask.google.com', release='5.10.0-smp-1107.69.0.0', version='#1 [v5.10.0-1107.69.0.0] SMP @1754001642', machine='x86_64')
XLA_PYTHON_CLIENT_PREALLOCATE=false
```

### PR DESCRIPTION
**Pull request by akshay-babbar**
Fixes #32300

The `local_window_size` parameter was being ignored when `mask=None` in `nn.dot_product_attention`. 

The `_apply_masks` function had an early return that only checked for `mask`, `is_causal`, `q_seqlen`, and `kv_seqlen`, but not `local_window_size`. This caused the function to skip masking entirely when no explicit mask was provided, even if a local window was specified.

**Changes:**
- Added `local_window_size is None` check to the early return condition in `_apply_masks`
- Added test `testDotProductAttention_localWindowSizeWithoutMask` to verify the fix

Thanks for reviewing!

### CODE REVIEW

**Comment by akshay-babbar** [comment]
hello @justinjfu, can you please review this PR and share feedback?

**Comment by akshay-babbar** [comment]
Hi @justinjfu, just wanted to check in if you’ve had a chance to review this PR. Thanks!

**Comment by akshay-babbar** [comment]
Hello @justinjfu , I understand you might be busy, no worries 🙂 
hello @jakevdp / @superbobry  , would you (or someone you recommend) be able to help review and merge this PR?
Thanks!

**Comment by akshay-babbar** [comment]
Hey @justinjfu, just checking in on this PR which has been open for a while (~ 2 months). Can we review this small PR and close this.
Will appreciate it!

Thanks!

**Review by superbobry — APPROVED** [review]
